uint8 iskeepCurSpeed = 1
uint8 notkeepCurSpeed = 0


float64 upperSpeed
float64 bottomSpeed
uint8 keepCurSpeed
